pkg/workload/workload.go

```diff
@@ -294,3 +294,16 @@ func AdmissionPatch(w *kueue.Workload) *kueue.Workload {
 	wlCopy.Status.Admission = w.Status.Admission.DeepCopy()
 	return wlCopy
 }
+
+// GetSchedulingTimestamp return the timestamp to be used by the scheduler. It could
+// be the workload creation time or the last time a PodsReady timeout has occurred.
+func GetSchedulingTimestamp(w *kueue.Workload) *metav1.Time {
+	evictedConditionPosition := FindConditionIndex(&w.Status, kueue.WorkloadEvicted)
```

Comment:
we should have gotten rid of `FindConditionIndex`. Let's use the metav1 helpers.